### PR TITLE
Allow PHP 8 and Update PHPUnit

### PR DESCRIPTION
**Pull request by mbonneau**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -50,4 +15,4 @@ script:
   - php docs/build-docs.php lint
 
 after_success:
-  - travis_retry php vendor/bin/coveralls -v
+  - travis_retry php vendor/bin/coveralls -v
```
Comment (asm89) [inline .travis.yml]: nit: revert?

File: docs/doc-loader.php [hunk 2]
```diff
@@ -77,7 +77,8 @@ public static function fromPath($path) {
         $outputPath = __DIR__ . '/../demo/' . $path . '.expect';
 
         assert(file_exists($codePath), "code does not exist $codePath");
-        assert(file_exists($outputPath), "output does not exist $outputPath");
+        // allow demos without expect files - we will just get a warning later
+        // assert(file_exists($outputPath), "output does not exist $outputPath");
```
Comment (asm89) [inline docs/doc-loader.php:81]: Why does this need to be removed? Perhaps we can skip and print a warning for now if the output path does not exist?

File: src/Scheduler/VirtualTimeScheduler.php [hunk 3]
```diff
@@ -21,7 +21,7 @@ class VirtualTimeScheduler implements AsyncSchedulerInterface
      * @param integer $initialClock Initial value for the clock.
      * @param callable $comparer Comparer to determine causality of events based on absolute time.
      */
-    public function __construct(int $initialClock = 0, callable $comparer)
+    public function __construct(int $initialClock, callable $comparer)
```
Comment (asm89) [inline src/Scheduler/VirtualTimeScheduler.php:24]: Interesting. I guess this went unnoticed before.

**Review by asm89 — APPROVED** [review]
LGTM!

File: docs/doc-loader.php (hunk 2 above)
Comment (mbonneau) [reply docs/doc-loader.php:81]: In PHP 7 this generated a warning and the tests still passed. With PHP 8 this generates a fatal error and stops.

A warning is still generated later when `file_get_contents` is called, but the tests still work.

File: .travis.yml (hunk 1 above)
Comment (mbonneau) [reply .travis.yml]: I don't remember removing that. I will take a look.

**Pull request merged**